### PR TITLE
Added tests for tags

### PR DESCRIPTION
**Pull request by Nyholm**
I missed a test. This test is taken from Symfony (https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Cache/Tests/Adapter/TagAwareAdapterTest.php)

FYI: @nicolas-grekas

### CODE REVIEW

File: src/TaggableCachePoolTest.php [hunk 1]
```diff
@@ -292,6 +275,20 @@ public function testInvalidateTags()
...
+     */
+    public function testTagsAreCleanedOnSave()
+    {
+        $pool = $this->cache;
+        $i    = $pool->getItem('key');
+        $pool->save($i->addTag('foo'));
+        $i = $pool->getItem('key');
+        $pool->save($i->addTag('bar'));
+        $pool->invalidateTags(['foo']);
+        $this->assertTrue($pool->getItem('key')->isHit());
```
Comment (Nyholm) [inline src/TaggableCachePoolTest.php:269]: Is this test correct? Or... is this the expected behavior? If I fetch an object from cache and then _add_ some tags to it. It should keep the existing tags, right?

The [Symfony test](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Cache/Tests/Adapter/TagAwareAdapterTest.php#L81) is using the `tag` function which has different features. 

I would say that this is something the tag PSR should specify. I want to change this test to: `$this->assertFalse($pool->getItem('key')->isHit());`

Comment (nicolas-grekas) [reply src/TaggableCachePoolTest.php:269]: IMHO, this is expected, see https://github.com/symfony/symfony/issues/19728#issuecomment-267920552 for a description of `getTags`' behavior.
Tags are an artifact of the *computation* of the cached data. Adding a tag without actually computing the value makes no sense in this respect.

Comment (nicolas-grekas) [reply src/TaggableCachePoolTest.php:269]: *when* it does have some sense for some particular need, then one could use first getTags, and use the result to set tags again. But this shouldn't be the default IMHO.

Comment (Nyholm) [reply src/TaggableCachePoolTest.php:269]: I moved this discussion to the https://github.com/symfony/symfony/issues/19728

**Pull request merged**